eng/native/gen-buildsys.sh

```diff
@@ -98,7 +98,7 @@ if [[ "$host_arch" == "wasm" ]]; then
     if [[ "$target_os" == "browser" ]]; then
         cmake_command="emcmake $cmake_command"
     elif [[ "$target_os" == "wasi" ]]; then
-        cmake_extra_defines="$cmake_extra_defines -DCLR_CMAKE_TARGET_OS=wasi -DCLR_CMAKE_TARGET_ARCH=wasm -DWASI_SDK_PREFIX=$WASI_SDK_PATH -DCMAKE_TOOLCHAIN_FILE=${WASI_SDK_PATH}share/cmake/wasi-sdk.cmake" "-DCMAKE_SYSROOT=${WASI_SDK_PATH}share/wasi-sysroot"
+        cmake_extra_defines=("$cmake_extra_defines" "-DCLR_CMAKE_TARGET_OS=wasi" "-DCLR_CMAKE_TARGET_ARCH=wasm" "-DWASI_SDK_PREFIX=$WASI_SDK_PATH" "-DCMAKE_TOOLCHAIN_FILE=$WASI_SDK_PATH/share/cmake/wasi-sdk.cmake" "-DCMAKE_SYSROOT=${WASI_SDK_PATH}share/wasi-sysroot" "-DCMAKE_CROSSCOMPILING_EMULATOR=node --experimental-wasm-bigint --experimental-wasi-unstable-preview1")
```

Comment:
Would you mind cherry-picking https://github.com/dotnet/runtimelab/pull/2614/commits/b2e8ff4819c4b2a5564b34409704b290610eab7f, since this won't work without that patch?

Also, per my comment on `gen-buildsys.cmd`, we should probably be consistent about how we select which `node` to use (i.e. decide whether to just always use whatever's in PATH or make it an environment variable).

Reply:
this is bad merge. I wanted p2 tool file, but not the extra `cmake_extra_defines_wasm`, let me try again.

Reply:
ok, I don't know how to do it without `cmake_extra_defines_wasm`, I take your version